### PR TITLE
Added communication with OpenSearch

### PR DESCRIPTION
**Pull request by owaiskazi19**
Added communication with OpenSearch done:

- Removed dummy DIscoveryNode
- Added old Netty4Transport to sync with OpenSearch code
- Added registerRequestHandler
- Used the default TransportService Constructor

### CODE REVIEW

**Review by joshpalis — APPROVED** [review]
Looks good to me :)

File: src/main/java/transportservice/RunPlugin.java [hunk 1]
```diff
@@ -17,21 +20,23 @@
...
 
 public class RunPlugin {
 
+    public static final String REQUEST_EXTENSION_ACTION_NAME = "internal:discovery/extensions";
     private static final Settings settings = Settings.builder()
-        .put("node.name", "NettySizeHeaderFrameDecoderTests")
+        .put("node.name", "node_extension")
         .put(TransportSettings.BIND_HOST.getKey(), "127.0.0.1")
-        .put(TransportSettings.PORT.getKey(), "9301")
+        .put(TransportSettings.PORT.getKey(), "4532")
```
Comment (saratvemulapalli) [inline src/main/java/transportservice/RunPlugin.java:39]: Lets get these properties from `extension.yml`. 
I know you have a PR, just link the issue :)

File: src/main/java/transportservice/RunPlugin.java [hunk 2]
```diff
@@ -70,18 +75,37 @@ private void startTransport() throws UnknownHostException {
             transport,
             threadPool,
             NOOP_TRANSPORT_INTERCEPTOR,
-            connectionManager,
+            boundAddress -> DiscoveryNode.createLocal(
+                Settings.builder().put("node.name", "node_extension").build(),
```
Comment (saratvemulapalli) [inline src/main/java/transportservice/RunPlugin.java:79]: Same here, set it globally/pass a variable. Lets keep it consistent across the SDK.

File: src/main/java/transportservice/RunPlugin.java [hunk 3]
```diff
@@ -70,18 +75,37 @@ private void startTransport() throws UnknownHostException {
...
         );
 
         transportService.start();
         transportService.acceptIncomingRequests();
+        transportService.registerRequestHandler(
+            REQUEST_EXTENSION_ACTION_NAME,
+            ThreadPool.Names.GENERIC,
+            false,
+            false,
+            PluginRequest::new,
```
Comment (saratvemulapalli) [inline src/main/java/transportservice/RunPlugin.java:95]: Yay!

**Review by saratvemulapalli — APPROVED** [review]
The changes look good to me. 
Thanks @owaiskazi19!

Few minor comments.

File: src/main/java/transportservice/RunPlugin.java (hunk 1 above)
Comment (owaiskazi19) [reply src/main/java/transportservice/RunPlugin.java:39]: Addressed in https://github.com/owaiskazi19/IndependentPlugin/pull/10

File: src/main/java/transportservice/RunPlugin.java (hunk 2 above)
Comment (owaiskazi19) [reply src/main/java/transportservice/RunPlugin.java:79]: Will push the change in https://github.com/owaiskazi19/IndependentPlugin/pull/10 after we merge this PR.

**Pull request merged**